### PR TITLE
Use fixed Root.arcs manifest name.

### PR DESCRIPTION
**Pull request by piotrswigon**
With this change, the root manifest will always be `/assets/arcs/Root.arcs`. This simplifies manifest setup, as the source code no longer needs to know what's the location and name of the root manifest.

This can be seen as a limitation if we wanted to generate multiple separate manifest sets - but this is not a requirement we have (and not something we even envision), so let's not complicate our lives if we don't need to.

This change also makes the Root.arcs manifest loaded from the APK even if the load_workstation_assets switch is on - this is because we don't foresee the root manifest changing dynamically during feature development and mapping it to the workstation location is non trivial, as it's location depends on the location of the arcs_manifest_bundle in the source tree.

### CODE REVIEW

File: javaharness/java/arcs/android/AndroidArcsEnvironment.java [hunk 1]
```diff
@@ -41,7 +38,9 @@
   private static final Logger logger =
     Logger.getLogger(AndroidArcsEnvironment.class.getName());
 
-  private static final String ASSETS_PREFIX = "https://$assets/";
+  private static final String ASSETS_MACRO = "https://$assets/";
```
Comment (csilvestrini) [inline javaharness/java/arcs/android/AndroidArcsEnvironment.java]: Macro? That name is a bit confusing to me. tbh I preferred ASSETS_PREFIX, or maybe ASSETS_PATH.

File: javaharness/java/arcs/android/AndroidArcsEnvironment.java [hunk 2]
```diff
@@ -41,7 +38,9 @@
...
+  private static final String APK_ASSETS_URL_PREFIX = "file:////android_asset/arcs/";
+  private static final String ROOT_MANIFEST_FILENAME = "Root.arcs";
```
Comment (csilvestrini) [inline javaharness/java/arcs/android/AndroidArcsEnvironment.java]: Super minor nit: `root.arcs` seems better to me :P

File: javaharness/java/arcs/android/ArcsService.java [hunk 3]
```diff
@@ -84,7 +84,8 @@ public void registerRenderer(String modality, IRemoteOutputCallback callback) {
 
       @Override
       public boolean addManifests(List<String> manifests) {
-        return arcsShellApi.addManifests(manifests);
+        // TODO: Implement once dynamic manifests are needed.
+        return true;
```
Comment (csilvestrini) [inline javaharness/java/arcs/android/ArcsService.java]: Could we do this instead? `throw new UnsupportedOperationException();`

That's way less surprising.

File: javaharness/java/arcs/android/AndroidArcsEnvironment.java (hunk 2 above)
Comment (piotrswigon) [reply javaharness/java/arcs/android/AndroidArcsEnvironment.java]: Manifest names were supposed to be Upper camel case:
https://docs.google.com/document/d/1QgvAX8M8s3g16zbICaK2NPqoTdma9dSxgJi6HnZRw3I/edit

File: javaharness/java/arcs/android/ArcsService.java (hunk 3 above)
Comment (piotrswigon) [reply javaharness/java/arcs/android/ArcsService.java]: Done.

File: javaharness/java/arcs/android/AndroidArcsEnvironment.java (hunk 1 above)
Comment (piotrswigon) [reply javaharness/java/arcs/android/AndroidArcsEnvironment.java]: Done.

**Comment by yuangu-google** [comment]
This approach makes it quite hard to support dynamic manifest in the future as the path is hard-coded and we only support load manifest once when the runtime starts.

With addManifest, it's doable as we can have some other service/activity downloads the manifest, add the downloaded ones along with this hard-coded root manifest, and pass it to ArcsService with addManifest.

My suggestion is that we either keep the addManifest and delegate the manifest loading work to other service/activity, or we remove it completely and only add it back when runtime supports dynamic loading manifests, i.e. not only loading once. Either way seems better than having an unsupported API.

**Comment by piotrswigon** [comment]
Thank you @yuangu-google. I've decided to remove the API completely for now. Let's re-add it with the support for both adding and removing manifests later on when we tackle dynamically changing set of manifests. For now let's focus on the manifests that are bundled at compile time into the APK.

**Pull request merged**